components/writers/src/main/java/org/datacleaner/extension/output/CreateCsvFileAnalyzer.java

```diff
@@ -155,10 +161,10 @@ public OutputWriter createOutputWriter() {
 
         List<String> headers = new ArrayList<String>();
         for (int i = 0; i < columns.length; i++) {
-            String columnName = columns[i].getName();
+            String columnName = getColumnHeader(i);
             headers.add(columnName);
             if (columnToBeSortedOn != null) {
-                if (columnName.equals(columnToBeSortedOn.getName())) {
+                if (columns[i].getName().equals(columnToBeSortedOn.getName())) {
```

Comment:
I would argue that we shouldn't compare names, but actual columns. I know this is not new in this PR, but I think it's bad anyway :-P